client/dashboard/sites/deployments/deployment-logs/deployment-logs-modal.tsx

```diff
@@ -0,0 +1,150 @@
+import { __ } from '@wordpress/i18n';
+import { useLocale } from '../../../app/locale';
+import { formatDate } from '../../../utils/datetime';
+import { BranchDisplay } from '../branch-display';
+import { DeploymentLogsEntry } from './deployment-logs-entry';
+import { deploymentRunLogsQuery } from './deployment-logs-queries';
+import { DeploymentLogsStatus } from './deployment-logs-status';
+import type { DeploymentRunWithDeploymentInfo } from '@automattic/api-core';
+
+interface DeploymentLogsModalProps {
+	onRequestClose: () => void;
+	deployment: DeploymentRunWithDeploymentInfo;
+	siteId: number;
+}
+
+export function DeploymentLogsModal( {
+	onRequestClose,
+	deployment,
+	siteId,
+}: DeploymentLogsModalProps ) {
+	const locale = useLocale();
+
+	const isDeploymentInFinalState = ( status: string ) => {
+		return [ 'success', 'failed', 'warnings' ].includes( status );
+	};
+
+	const isFinalState = isDeploymentInFinalState( deployment.status );
+	const {
+		data: logEntries = [],
+		isLoading,
+		isError,
+	} = useQuery( {
+		...deploymentRunLogsQuery( siteId, deployment.code_deployment_id, deployment.id ),
+		refetchInterval: ! isFinalState ? 1000 : undefined,
+	} );
+
+	const { commit_message, commit_sha, author } = deployment.metadata;
+	const shortSha = commit_sha?.substring( 0, 7 ) || '';
+
+	return (
+		<Modal
+			title={ `${ shortSha } ${ commit_message }` }
+			onRequestClose={ onRequestClose }
+			className="deployment-logs-modal"
+			shouldCloseOnClickOutside
+			shouldCloseOnEsc
+		>
+			<HStack spacing={ 3 } alignment="left" style={ { width: '100%', marginBottom: '16px' } }>
+				{ deployment.is_active_deployment && (
+					<Badge style={ { flexShrink: 0 } }>{ __( 'Active deployment' ) }</Badge>
+				) }
+
+				<div style={ { width: 'auto', flexShrink: 0, maxWidth: '33vw' } }>
+					<BranchDisplay branchName={ deployment.branch_name } />
+				</div>
+
+				<HStack spacing={ 1.5 } alignment="left" style={ { width: 'auto', flexShrink: 0 } }>
+					<img
+						src={ author.avatar_url }
+						alt={ author.name }
+						width={ 16 }
+						height={ 16 }
+						style={ { borderRadius: '50%' } }
+					/>
+					<Text size="small" style={ { color: '#3b3b3b' } }>
+						{ author.name }
+					</Text>
+				</HStack>
+
+				<DeploymentLogsStatus
+					status={ deployment.status }
+					startedOn={ deployment.started_on }
+					completedOn={ deployment.completed_on }
+				/>
+			</HStack>
+
+			{ isError && <Text>{ __( 'Failed to load deployment logs. Please try again.' ) }</Text> }
+			{ ! isLoading && isDeploymentInFinalState( deployment.status ) && logEntries.length === 0 && (
+				<Text>{ __( 'No logs available for this deployment.' ) }</Text>
+			) }
+
+			{ isLoading && (
+				<HStack alignment="center">
+					<Spinner />
+				</HStack>
+			) }
+
+			{ ! isLoading &&
+				( logEntries.length > 0 || ! isDeploymentInFinalState( deployment.status ) ) && (
+					<VStack spacing={ 2 }>
+						<div
+							style={ {
+								width: '752px',
```

Comment:
It would be better to pass `size="large"` to the modal instead of adding width here.

Reply:
Done! It works perfectly. https://github.com/Automattic/wp-calypso/pull/105747/commits/b7bf1d67609642b57f010e6978209d6fdcac2e22